app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -363,6 +363,15 @@ protected void onCreate(Bundle savedInstanceState) {
 
         GeolocationWrapper.INSTANCE.update(this);
 
+        try {
+            String speechService = SettingsStore.getInstance(this).getVoiceSearchService();
+            SpeechRecognizer speechRecognizer = SpeechServices.getInstance(this, speechService);
+            ((VRBrowserApplication) getApplicationContext()).setSpeechRecognizer(speechRecognizer);
+        } catch (Exception e) {
+            Log.e(LOGTAG, "Exception creating the speech recognizer: " + e);
+            e.printStackTrace();
+        }
```

Comment:
Any special reason to move this here?

Reply:
It needs to happen after the API key has been set, which happens on the method `PlatformActivity.onCreate()`.